infrastructure/src/main/java/org/corfudb/infrastructure/health/HealthReport.java

```diff
@@ -0,0 +1,195 @@
+package org.corfudb.infrastructure.health;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.annotations.SerializedName;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.ToString;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.corfudb.infrastructure.health.HealthReport.ComponentStatus.DOWN;
+import static org.corfudb.infrastructure.health.HealthReport.ComponentStatus.FAILURE;
+import static org.corfudb.infrastructure.health.HealthReport.ComponentStatus.UNKNOWN;
+import static org.corfudb.infrastructure.health.HealthReport.ComponentStatus.UP;
+
+/**
+ * HealthReport represents the overall health status of the node.
+ */
+@Builder
+@ToString
+@EqualsAndHashCode
+public class HealthReport {
+
+    /**
+     * Overall status msgs
+     */
+    public static final String OVERALL_STATUS_UNKNOWN = "Status is unknown";
+    public static final String OVERALL_STATUS_UP = "Healthy";
+    public static final String OVERALL_STATUS_DOWN = "Some of the services are not initialized";
+    public static final String OVERALL_STATUS_FAILURE = "Some of the services experience runtime health issues";
+
+    /**
+     * Component status msgs
+     */
+    public static final String COMPONENT_INITIALIZED = "Initialization successful";
+    public static final String COMPONENT_NOT_INITIALIZED = "Service is not initialized";
+    public static final String COMPONENT_IS_NOT_RUNNING = "Service is not running";
+    public static final String COMPONENT_IS_RUNNING = "Up and running";
+
+    /**
+     * Health status false = not healthy, true = healthy
+     */
+    @Builder.Default
+    @Getter
+    private final ComponentStatus status = UNKNOWN;
+    /**
+     * Description
+     */
+    @Builder.Default
+    @Getter
+    private final String reason = OVERALL_STATUS_UNKNOWN;
+
+    /**
+     * Set of component initialization health issues
+     */
+    @NonNull
+    @Getter
+    private final Set<ComponentReportedHealthStatus> init;
```

Comment:
nit: would it be better to rename these as 'initHealthIssues' and 'runtimeHealthIssues' instead?

Reply:
fixed